packages/react/src/Tabs/Tab/useTab.ts

```diff
@@ -56,28 +86,52 @@ function useTab(parameters: useTab.Parameters): useTab.ReturnValue {
             'aria-selected': selected,
             id,
             ref: handleRef,
+            onClick(event) {
+              onTabActivation(tabValue, event.nativeEvent);
+            },
+            onFocus(event) {
+              if (!activateOnFocus) {
+                return;
+              }
+
+              if (selectedTabValue !== tabValue) {
+                onTabActivation(tabValue, event.nativeEvent);
+              }
+            },
           },
-          mergeReactProps(getListItemProps(), getButtonProps()),
+          mergeReactProps(getItemProps(), getButtonProps()),
         ),
       );
     },
-    [getButtonProps, getListItemProps, handleRef, id, selected, tabPanelId],
+    [
+      activateOnFocus,
+      getButtonProps,
+      getItemProps,
+      handleRef,
+      id,
+      onTabActivation,
+      selected,
+      selectedTabValue,
+      tabPanelId,
+      tabValue,
+    ],
   );
 
   return {
     getRootProps,
     index,
     rootRef: handleRef,
-    // the `selected` state isn't set on the server (it relies on effects to be calculated),
-    // so we fall back to checking the `value` prop with the selectedValue from the TabsContext
-    selected: selected || value === selectedValue,
-    totalTabsCount,
-    orientation,
+    selected,
+    // TODO: recalculate this using Composite stuff, but is it really needed?
+    totalTabsCount: -1,
```

Comment:
@michaldudak Is it ok to drop this? I found it wasn't actually used in the Tab component